contents/df_end.md

```diff
@@ -141,36 +150,287 @@ filter(:name => ==("Alice"), grades_2020())
 To get all the rows which are **not** Alice, `==` can be replaced by `!=` in all previous examples:
 
 ```jl
-sco("""
-filter(:name => !=("Alice"), grades_2020())
-"""; process=without_caption_label)
+s = """filter(:name => !=("Alice"), grades_2020())"""
+sco(s; process=without_caption_label)
 ```
 
 Now, to show why anonymous functions are so powerful, we can come up with a more complex filter.
 In this filter, we want to have the people whos name start with A or B **and** who have a grade above a 6:
 
 ```jl
-sc("""
-function complex_filter(name, grade)::Bool
-    interesting_name = startswith(name, 'A') || startswith(name, 'B')
-    interesting_grade = 6 < grade
-    interesting_name && interesting_grade
-end
-""")
+s = """
+    function complex_filter(name, grade)::Bool
+        interesting_name = startswith(name, 'A') || startswith(name, 'B')
+        interesting_grade = 6 < grade
+        interesting_name && interesting_grade
+    end
+    """
+sc(s)
 ```
 
 ```jl
-sco("""
-filter([:name, :grade_2020] => complex_filter, grades_2020())
-"""; process=without_caption_label)
+s = "filter([:name, :grade_2020] => complex_filter, grades_2020())"
+sco(s; process=without_caption_label)
+```
+
+### Subset {#sec:subset}
+
+The `subset` function was added to make it easier to work with missing values (@sec:missing_data).
+In contrast to `filter`, the `subset` function works on complete columns.
+If we want to use our earlier defined functions, we can use `ByRow`:
+
+```jl
+s = "subset(grades_2020(), :name => ByRow(equals_alice))"
+sco(s; process=without_caption_label)
+```
+
+Also note that the DataFrame is now the first argument, whereas it was the second argument in `filter`, that is, use `filter(f, df)` and use `subset(df, args...)`.
+The reason for this is that Julia defines filter as `filter(f, V::Vector)` and the developers of `DataFrames.jl` chose to be consistent with that.
+
+Just like with `filter`, we can also use anonymous functions:
+
+```jl
+s = "subset(grades_2020(), :name => ByRow(name -> name == \"Alice\"))"
+sco(s; process=without_caption_label)
+```
+
+Or, the partial function application for `==`:
+
+```jl
+s = "subset(grades_2020(), :name => ByRow(==(\"Alice\")))"
+sco(s; process=without_caption_label)
+```
+
+To, now, show the real power of `subset`, let's create a dataset with some missing values:
+
+```jl
+@sco salaries()
+```
+
+This data is about an imaginary situation where you want to figure out how many your colleagues earn, and haven't figured it out for Zed yet.
```

Comment:
```suggestion
This data is about a plausible situation where you want to figure out how many your colleagues earn, and haven't figured it out for Zed yet.
```

who is "Zed"?

Reply:
You'll see when I merge and the site is updated :stuck_out_tongue: